backend/app/assets/javascripts/spree/backend/product_picker.js

```diff
@@ -1,9 +1,32 @@
+/**
+  * Make the element a select2 dropdown used for finding Products. By default,
+  * it allows the Products to be found by its name and its Variants' SKUs.
+  * @param  {Object} options Options
+  * @param  {Boolean} [options.multiple=true] Allow multiple products to be selectable
+  * @param  {Function|undefined} options.searchCallback Returns a hash object for params to merge on the select2 ajax request
```

Comment:
Nice work on the documentation, this is 💯 

I am not very happy with the name, though. It is very generic and too technical imo. Sure, it is - technically - a callback, but it is not very descriptive about what it is used for or what it shall return. What about:

- `searchParameters` - My favorite
- `extraParameters` (would fit to the function it is used in)

Reply:
Understood. I found the naming difficult. I do agree that `searchCallback` is not perfect, though to me `searchParameters` sounds like it should be a hash object rather than a function that should return a hash? 🤔 @tvdeyen

Reply:
On further thoughts, the documentation does show `Function`. So I think the suggestions are acceptable. Note, I made it so the function's returning object is extended onto the request parameters, so it doesn't have to just be just the ransack `q` param. Interested on your thoughts @waiting-for-dev

Reply:
`searchCallback` changed to `searchParameters`